lib/matplotlib/tests/test_legend.py

```diff
@@ -532,6 +532,32 @@ def test_empty_bar_chart_with_legend():
     plt.legend()
 
 
+@image_comparison(['shadow_argument_types.png'])
+def test_shadow_argument_types():
+    # Test that different arguments for shadow work as expected
+    fig, ax = plt.subplots()
+    ax.plot([1, 2, 3], label='test')
+
+    legs = (ax.legend(loc='upper left', shadow=True),                          # True
+            ax.legend(loc='upper right', shadow=False),                        # False
+            ax.legend(loc='center left', shadow={'color': 'red'}),             # string
+            ax.legend(loc='center right', shadow={'color': (0.1, 0.2, 0.5)}),  # tuple
+            ax.legend(loc='lower left', shadow={'color': 'tab:cyan'})          # tab
```

Comment:
I'd suggest changing ox and oy here as well for some of the tests, possibly also some other Shadow parameters.

Reply:
I added changing `ox, oy` and `alpha`, but when the test was failing in generating the baseline image for the first time, the diff didn't seem to notice the difference in `alpha`. This was before rebasing, but is that a concern?

Reply:
Not sure I fully understand, but I think the new tests cover most of the things, so it should be OK! (alpha on white may not always give a very clear effect, so it may be that it is not distinguishable.)